crates/crypto_api/src/buffer/buffer_vec_u8.rs

```diff
@@ -19,4 +19,21 @@ impl Buffer for Vec<u8> {
     fn set_no_access(&self) {}
     fn set_readable(&self) {}
     fn set_writable(&self) {}
+
+    fn compare(&mut self, b: &mut Box<dyn Buffer>) -> i32 {
+        let l = self.len();
+        if l != b.len() {
+            return 1;
+        }
+        let a = self.read_lock();
+        let b = b.read_lock();
+        let mut i = 0;
+        while i < l {
+            if a[i] != b[i] {
+                return 1;
```

Comment:
We actually need this to return -1 if a < b when interpreted as little endian per https://download.libsodium.org/doc/helpers#comparing-large-numbers.

We could potentially use https://docs.rs/libc/0.2.63/libc/fn.memcmp.html (though I can't remember, we might need to reverse the result)

Reply:
code now compares values, but probably doesn't do the right thing on different length buffers, but at lease I added a test.  @neonphog can you confirm what the appropriate expected behavior of compare should be?